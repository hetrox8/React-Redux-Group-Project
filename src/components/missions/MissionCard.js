import React from 'react';
import PropTypes from 'prop-types';
import Badge from 'react-bootstrap/Badge';
import JoinMission from './JoinMission';

const MissionCard = (props) => {
  const { mission } = props;
  const switchBadge = () => {
    if (mission.isJoined) {
      return <Badge bg="info">Active Member</Badge>;
    }
    return <Badge bg="secondary">Not A Member</Badge>;
  };

  return (
    <tr>
      <th>
        <h5>{mission.mission_name}</h5>
      </th>
      <th className="description">
        <p>{mission.mission_description}</p>
      </th>
      <th className="mission-status">
        {switchBadge()}
        <div>
          <JoinMission
            isJoined={mission.isJoined}
            id={mission.mission_id}
          />
        </div>
      </th>
    </tr>
  );
};

MissionCard.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string,
    mission_name: PropTypes.string,
    mission_description: PropTypes.string,
    isJoined: PropTypes.bool,
  }).isRequired,
};

export default MissionCard;
