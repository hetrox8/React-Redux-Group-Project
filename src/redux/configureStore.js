import {
  legacy_createStore as createStore,
  applyMiddleware,
  combineReducers,
} from 'redux';
import thunk from 'redux-thunk';
import rocketsReducer from './rocket/Rocket';

const rootReducer = combineReducers({
  rocket: rocketsReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
