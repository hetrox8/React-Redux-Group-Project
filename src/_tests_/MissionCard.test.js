import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MissionCard from '../components/missions/MissionCard';

const mockStore = configureStore([]);

describe('MissionCard', () => {
  test('MissionCard rendered correctly', () => {
    const initialState = {
      missions: [
        {
          name: 'Test Mission',
          mission_description: 'Test Description',
          isJoined: false,
          mission_id: 'test-id',
        },
      ],
    };

    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <MissionCard mission={store.getState().missions[0]} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
