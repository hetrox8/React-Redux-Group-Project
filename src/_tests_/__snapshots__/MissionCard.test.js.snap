// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionCard MissionCard rendered correctly 1`] = `
<DocumentFragment>
  <h5 />
  <p>
    Test Description
  </p>
  <span
    class="badge bg-secondary"
  >
    Not A Member
  </span>
  <div>
    <button
      class="btn btn-outline-primary"
      type="button"
    >
      Join  Mission
    </button>
  </div>
</DocumentFragment>
`;
